### PR TITLE
Remove WindowManager reflection in SingleViewPresentation.java

### PR DESCRIPTION
**Pull request by gmackall**
Fixes https://github.com/flutter/flutter/issues/106449. Changes it to a static proxy, as the comment recommended. This does mean we will have to update it to override new methods as they are added to the interface when updating the version of the Android sdk we use.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/android/io/flutter/plugin/platform/SingleViewPresentation.java [hunk 1]
```diff
@@ -377,21 +377,17 @@ private boolean isCalledFromAlertDialog() {
...
    * WebView (as the selection handles are implemented as popup windows).
    *
-   * This dynamic proxy overrides the addView, removeView, removeViewImmediate, and updateViewLayout methods
-   * to prevent these crashes.
-   *
-   * This will be more efficient as a static proxy that's not using reflection, but as the engine is currently
-   * not being built against the latest Android SDK we cannot override all relevant method.
-   * Tracking issue for upgrading the engine's Android sdk: https://github.com/flutter/flutter/issues/20717
+   * This static proxy overrides the addView, removeView, removeViewImmediate, and updateViewLayout methods
+   * to prevent these crashes, and passes all other calls to the delegate.
```
Comment (johnmccutchan) [inline shell/platform/android/io/flutter/plugin/platform/SingleViewPresentation.java:388]: s/passes/forwards

**Review by johnmccutchan — APPROVED** [review]
Thanks for this cleanup!

Can you confirm that this code is covered by an existing test or add a test to the PR?

**Comment by chinmaygarde** [comment]
Can we land this?

**Comment by gmackall** [comment]
> Can we land this?

Sorry, mentioned to @johnmccutchan offline but didn't add a comment - I need to determine if the existing test is in fact testing this. I'll figure that out today

**Comment by gmackall** [comment]
It looks like the proxy was originally added in https://github.com/flutter/engine/pull/6043, which did not include tests. I'll need to write a test to cover.

**Comment by chinmaygarde** [comment]
Sounds good. Marking this as WIP so we don't bother you with triage.

**Comment by gmackall** [comment]
Bonus: I can now run these tests straight from Android Studio with just a click, just need to clean up https://github.com/flutter/engine/compare/main...gmackall:engine:android_java_unittests_in_AS and land it and others will be able to as well

**Review by reidbaker — APPROVED** [review]

**Review by johnmccutchan — APPROVED** [review]

**Comment by gmackall** [comment]
Reason for revert: b/326363243

**Pull request merged**